pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -486,7 +490,16 @@ func (c *ruleCache) AddAppliedToGroup(group *v1beta.AppliedToGroup) error {
 func (c *ruleCache) addAppliedToGroupLocked(group *v1beta.AppliedToGroup) error {
 	memberSet := v1beta.GroupMemberSet{}
 	for i := range group.GroupMembers {
-		memberSet.Insert(&group.GroupMembers[i])
+		m := &group.GroupMembers[i]
+		if c.appliedToExternalEntity && m.Pod == nil && m.ExternalEntity != nil {
+			// Convert ExternalEntity to PodEntity so that rules applied to ExternalEntity.
```

Comment:
Having separate knobs for Antrea-native policy features sounds fine by me. However, I think converting externalEntityReference to podReference here is too much a hack imo. If we know that ExternalEntities will be configured with container interfaces just like Pods, we could refactor `GetContainerInterfacesByPod()` method to be more generic and be applicable to EEs as well. The same goes for `getPodOFPorts` and `getPodIPs`. @jianjuns @tnqn

Reply:
I would agree what Yang said. We could revise the struct or add a new to support ExternalEntity.

Reply:
Agree with @Dyanngg, I think that's also why we replaced GroupMemberPodSet with GroupMemberSet in many places.

Reply:
`appliedToExternalEntity` is not really needed, right? If there's ExternalEntity in the group, I think there's no reason to ignore it?

Reply:
I took approach with the least change/resistence to allow appliedToExternalEntity supportability,  as I am not very familiar with the implementation, and rather not to making sweeping changes.

In my view, if to address this properly, there are two approaches:
1. Using AppliedToGroup CRD ( from antrea manifest) as cue to support appliedToExternalEntity or not in antrea. On agent side we will support  AppliedToExternalEntity based on content.
2. In addition to 1, to be explicit , so that user cannot accidentally change AppliedToExternalEntity supportability by change antrea manifest. An compilation explicit flag is needed on agent side to indicate supportability.

@jianjuns @tnqn @Dyanngg what do u think? 

And if we do choose of the two above approaches, wonder if I get some help from u guys. It will definitely much efficient if @Dyanngg or @tnqn has a look?

Reply:
Hi @jianjuns @tnqn @Dyanngg need ur input on how to proceed with this PR.

Reply:
I am not sure if I got your #2, but definitely I would agree with your #1 that Agent should support ExternalEntity. But in your mind, if Agent does not really implements NetworkPolicy for ExternalEntity, what will be the interface it exposes to you? Probably we should discuss offline to understand your requirements.